### PR TITLE
Improving engagements

### PR DESCRIPTION
**Pull request by berCamargos**
# Problem
Wee have an infinite amount of engagements to get all in a single memory would be a really bad idea

# Solution
We added a generator, to allow the program to store to disk or send them in the desired quantity

# Obs
The actual code is old, I just did the PR right now

### CODE REVIEW

File: hubspot3/engagements.py [hunk 1]
```diff
@@ -64,21 +64,39 @@ def update(self, key, data=None, **options):
             "engagements/{}".format(key), data=data, method="PUT", **options
         )
 
-    def get_all(self, **options):
+    def get_all(self, limit=-1, **options):
+        engagement_generator = self.get_all_as_generator(limit=limit,
+                                                         **options)
+        return list(engagement_generator)
+
+    def get_all_as_generator(self, limit=-1, **options):
```
Comment (lsflp) [inline hubspot3/engagements.py:72]: I think you could do `limit=None` here, and then check if it exists by line 76.

File: hubspot3/engagements.py [hunk 2]
```diff
@@ -64,21 +64,39 @@ def update(self, key, data=None, **options):
             "engagements/{}".format(key), data=data, method="PUT", **options
         )
 
-    def get_all(self, **options):
+    def get_all(self, limit=-1, **options):
+        engagement_generator = self.get_all_as_generator(limit=limit,
```
Comment (lsflp) [inline hubspot3/engagements.py:68]: Do we have extra_properties for engagements? Even if we are not going to use them, the could should be able to handle them.

Comment (berCamargos) [reply hubspot3/engagements.py:68]: There are no property choices in the engagements, the only options are items per page and offset

https://developers.hubspot.com/docs/methods/engagements/get-all-engagements

File: hubspot3/engagements.py (hunk 1 above)
Comment (berCamargos) [reply hubspot3/engagements.py:72]: I agree that limit=None is better, but this form was already defined there and is used in all the objects I think it is better to keep it consistent through the code, what do you think?

Comment (lsflp) [reply hubspot3/engagements.py:72]: OK, I keep forgetting that this code is a little _weird_.

File: hubspot3/engagements.py (hunk 2 above)
Comment (lsflp) [reply hubspot3/engagements.py:68]: Great!

**Pull request merged**